src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,63 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
 
+    public MateStack() {
+        size = 0;
+    }
+
+    public void push(T value) {
+        if (size == 0) {
+            head = new Node<>(value, null);
+            tail = head;
+            size++;
+            return;
+        }
+        tail.next = new Node<>(value, null);
+        tail = tail.next;
+        size++;
```

Comment:
This can be simplified if we imagine our stack as a singly list (see task description and previous comment)